index.js

```diff
@@ -1579,6 +1579,76 @@ class GithubScm extends Scm {
             throw err;
         });
     }
+
+    /**
+    * Open a pull request on the repository with given file change
+    *
+    * @method _openPr
+    * @param  {Object}     config                  Configuration
+    * @param  {String}     config.scmUri           SCM URI to open pull request on
+    * @param  {String}     config.token            Service token to authenticate with the SCM service
+    * @param  {String}     config.files            Files to open pull request with
+    * @param  {String}     config.title            Pull request title
+    * @param  {String}     config.message          Pull request message
+    * @param  {String}     [config.scmContext]     The scm context name
+    * @return {Promise}                            Resolves when operation completed without failure
+    */
+    async _openPr(config) {
+        const { scmUri, token, files, title, message } = config;
+        const branch = 'master'; // todo: default to master?
+        const SCM_URL_REGEX = /^(?:(?:https?|git):\/\/)?(?:[^@]+@)?([^/:]+)(?:\/|:)([^/]+)\/(.+?)(?:\.git)?(#.+)?$/;
```

Comment:
should probably use data-schema https://github.com/screwdriver-cd/data-schema/blob/master/config/regex.js#L83

Reply:
Second to that. Use the same regex that `checkoutUrl` uses